## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-erasure.sh | :x: [more...](#61d7c8b5-3ef5-4c6b-92b4-0340ce149533) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#303d430c-0c47-44a0-b52a-dcd71301d6af) |
| mint-pools.sh | :x: [more...](#0065249e-850b-479b-ab9a-81730caefe32) |
<a name="303d430c-0c47-44a0-b52a-dcd71301d6af"></a>
### 16665-88f693f/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.6:30168
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 7d708cfd1393:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 1 minutes and 14 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 17 seconds
(8/14) Running minio-go tests ... done in 49 seconds
(9/14) Running minio-java tests ... done in 31 seconds
(10/14) Running minio-js tests ... FAILED in 1 minutes and 26 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 14,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(10/14) Running minio-py tests ... done in 2 minutes and 4 seconds
(11/14) Running s3cmd tests ... done in 15 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 18 seconds

Executed 13 out of 14 tests successfully.
```
<a name="0065249e-850b-479b-ab9a-81730caefe32"></a>
### 16665-88f693f/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.7:30031
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp ea5945a06c49:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 39 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 1 minutes and 18 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 18 seconds
(8/14) Running minio-go tests ... FAILED in 2 minutes and 3 seconds
{
  "args": {},
  "duration": 63529,
  "error": "The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.",
  "function": "CopyObjectPart(destination, source)",
  "message": "CompleteMultipartUpload call failed",
  "name": "minio-go: testSSES3EncryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/14) Running minio-java tests ... FAILED in 1 minutes and 0 seconds
{
  "name": "minio-java",
  "function": "removeBucket()",
  "args": "[basic check]",
  "duration": 57908,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketAlreadyOwnedByYou, message = Your previous request to create the named bucket succeeded and you already own it., bucketName = minio-java-test-28ogq6h, objectName = null, resource = /minio-java-test-28ogq6h, requestId = 174610C219B8D3FE, hostId = e6cea30f-1fee-4600-b829-f0c5289338cc)\nrequest={method=PUT, url=http://15.15.15.7:30031/minio-java-test-28ogq6h, headers=Host: 15.15.15.7:30031\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.6\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20230222T061119Z\nAuthorization: ██\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 376\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Id-2: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nX-Amz-Request-Id: 174610C219B8D3FE\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Wed, 22 Feb 2023 06:12:17 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(8/14) Running minio-js tests ... done in 53 seconds
(9/14) Running minio-py tests ... done in 4 minutes and 12 seconds
(10/14) Running s3cmd tests ... done in 18 seconds
(11/14) Running s3select tests ... done in 4 seconds
(12/14) Running versioning tests ... done in 3 minutes and 34 seconds

Executed 12 out of 14 tests successfully.
```
<a name="61d7c8b5-3ef5-4c6b-92b4-0340ce149533"></a>
### 16665-88f693f/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.4:30356
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 42491cdf1a3a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 8 seconds
(5/14) Running awscli tests ... done in 1 minutes and 13 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 14 seconds
(8/14) Running minio-go tests ... done in 35 seconds
(9/14) Running minio-java tests ... done in 18 seconds
(10/14) Running minio-js tests ... done in 46 seconds
(11/14) Running minio-py tests ... FAILED in 1 minutes and 2 seconds
{
  "name": "minio-py:test_put_object",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-167527a6-8a58-46f1-bce9-474d717d70e1",
    "object_name": "46e89189-fbfd-451f-ba63-3d7a752cefef-metadata",
    "length": 11534336,
    "data": "LimitedRandomReader(11 * MB)",
    "metadata": {
      "x-amz-meta-testing": "value",
      "test-key": "value2"
    },
    "content_type": "application/octet-stream"
  },
  "message": "('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 449, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 3, in raise_from\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 444, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.8/http/client.py\", line 1348, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.8/http/client.py\", line 316, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.8/http/client.py\", line 277, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n  File \"/usr/lib/python3.8/socket.py\", line 669, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 697, in test_put_object\n    _CLIENT.put_object(bucket_name, object_name + \"-metadata\", reader,\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1766, in put_object\n    raise exc\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1725, in put_object\n    upload_id = self._create_multipart_upload(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 1565, in _create_multipart_upload\n    response = self._execute(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 403, in _execute\n    return self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 550, in increment\n    raise six.reraise(type(error), error, _stacktrace)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/packages/six.py\", line 769, in reraise\n    raise value.with_traceback(tb)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 449, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 3, in raise_from\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 444, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.8/http/client.py\", line 1348, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.8/http/client.py\", line 316, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.8/http/client.py\", line 277, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n  File \"/usr/lib/python3.8/socket.py\", line 669, in readinto\n    return self._sock.recv_into(b)\nurllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))\n",
  "duration": 61220
}
(11/14) Running s3cmd tests ... done in 16 seconds
(12/14) Running s3select tests ... done in 3 seconds
(13/14) Running versioning tests ... done in 3 minutes and 1 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/admin-handlers-config-kv.go
+++ b/cmd/admin-handlers-config-kv.go
@@ -29,13 +29,13 @@ import (
 	"github.com/minio/madmin-go/v2"
 	"github.com/minio/minio/internal/config"
 	"github.com/minio/minio/internal/config/cache"
-	"github.com/minio/minio/internal/config/callhome"
 	"github.com/minio/minio/internal/config/etcd"
 	xldap "github.com/minio/minio/internal/config/identity/ldap"
 	"github.com/minio/minio/internal/config/identity/openid"
 	idplugin "github.com/minio/minio/internal/config/identity/plugin"
 	polplugin "github.com/minio/minio/internal/config/policy/plugin"
 	"github.com/minio/minio/internal/config/storageclass"
+	"github.com/minio/minio/internal/config/subnet"
 	"github.com/minio/minio/internal/logger"
 	"github.com/minio/mux"
 	iampolicy "github.com/minio/pkg/iam/policy"
@@ -180,7 +180,7 @@ func (a adminAPIHandlers) SetConfigKVHandler(w http.ResponseWriter, r *http.Requ
 		// If logger webhook config updated (proxy due to callhome), explicitly dynamically
 		// apply the config
 		if result.LoggerWebhookCfgUpdated {
-			applyDynamic(ctx, objectAPI, result.Cfg, "logger_webhook", r, w)
+			applyDynamic(ctx, objectAPI, result.Cfg, config.LoggerWebhookSubSys, r, w)
 		}
 	}
 
@@ -208,14 +208,13 @@ func setConfigKV(ctx context.Context, objectAPI ObjectLayer, kvBytes []byte) (re
 		return
 	}
 
-	// Check if callhome getting enabled, if so set the subnet proxy value to logger_webhook's subnet proxy
-	// Do this only if logger_wenhook's subnet proxy is not yet set
-	callhomeCfg := result.Cfg[config.CallhomeSubSys]["_"]
-	if result.SubSys == config.CallhomeSubSys && callhomeCfg.Get(callhome.Enable) == "on" {
+	// Check if subnet proxy being set and if so set the same value to proxy of subnet
+	// target of logger webhook configuartion
+	if result.SubSys == config.SubnetSubSys {
 		loggerWebhookSubnetProxy := result.Cfg[config.LoggerWebhookSubSys][config.SubnetSubSys].Get(logger.Proxy)
 		subnetProxy := result.Cfg[config.SubnetSubSys]["_"].Get(logger.Proxy)
-		if loggerWebhookSubnetProxy == "" && subnetProxy != "" {
-			obj := result.Cfg[config.LoggerWebhookSubSys][config.SubnetSubSys]
+		if loggerWebhookSubnetProxy != subnetProxy {
+			obj := result.Cfg[config.LoggerWebhookSubSys][subnet.LoggerWebhookName]
 			obj.Set(logger.Proxy, subnetProxy)
 			result.LoggerWebhookCfgUpdated = true
 		}

--- a/cmd/admin-handlers-config-kv.go
+++ b/cmd/admin-handlers-config-kv.go
@@ -132,9 +132,10 @@ func (bce badConfigErr) Unwrap() error {
 }
 
 type setConfigResult struct {
-	Cfg     config.Config
-	SubSys  string
-	Dynamic bool
+	Cfg                     config.Config
+	SubSys                  string
+	Dynamic                 bool
+	LoggerWebhookCfgUpdated bool
 }
 
 // SetConfigKVHandler - PUT /minio/admin/v3/set-config-kv
@@ -175,6 +176,11 @@ func (a adminAPIHandlers) SetConfigKVHandler(w http.ResponseWriter, r *http.Requ
 
 	if result.Dynamic {
 		applyDynamic(ctx, objectAPI, result.Cfg, result.SubSys, r, w)
+		// If logger webhook config updated (proxy due to callhome), explicitly dynamically
+		// apply the config
+		if result.LoggerWebhookCfgUpdated {
+			applyDynamic(ctx, objectAPI, result.Cfg, "logger_webhook", r, w)
+		}
 	}
 
 	writeSuccessResponseHeadersOnly(w)
@@ -199,6 +205,18 @@ func setConfigKV(ctx context.Context, objectAPI ObjectLayer, kvBytes []byte) (re
 	if verr := validateConfig(result.Cfg, result.SubSys); verr != nil {
 		err = badConfigErr{Err: verr}
 		return
+	}
+
+	// Check if callhome getting enabled, if so set the subnet proxy value to logger_webhook's subnet proxy
+	// Do this only if logger_wenhook's subnet proxy is not yet set
+	if result.SubSys == "callhome" && result.Cfg["callhome"]["_"].Get("enable") == "on" {
+		loggerWebhookSubnetProxy := result.Cfg["logger_webhook"]["subnet"].Get("proxy")
+		subnetProxy := result.Cfg["subnet"]["_"].Get("proxy")
+		if loggerWebhookSubnetProxy == "" && subnetProxy != "" {
+			obj := result.Cfg["logger_webhook"]["subnet"]
+			obj.Set("proxy", subnetProxy)
+			result.LoggerWebhookCfgUpdated = true
+		}
 	}
 
 	// Update the actual server config on disk.

--- a/cmd/admin-handlers-config-kv.go
+++ b/cmd/admin-handlers-config-kv.go
@@ -29,6 +29,7 @@ import (
 	"github.com/minio/madmin-go/v2"
 	"github.com/minio/minio/internal/config"
 	"github.com/minio/minio/internal/config/cache"
+	"github.com/minio/minio/internal/config/callhome"
 	"github.com/minio/minio/internal/config/etcd"
 	xldap "github.com/minio/minio/internal/config/identity/ldap"
 	"github.com/minio/minio/internal/config/identity/openid"
@@ -209,12 +210,13 @@ func setConfigKV(ctx context.Context, objectAPI ObjectLayer, kvBytes []byte) (re
 
 	// Check if callhome getting enabled, if so set the subnet proxy value to logger_webhook's subnet proxy
 	// Do this only if logger_wenhook's subnet proxy is not yet set
-	if result.SubSys == "callhome" && result.Cfg["callhome"]["_"].Get("enable") == "on" {
-		loggerWebhookSubnetProxy := result.Cfg["logger_webhook"]["subnet"].Get("proxy")
-		subnetProxy := result.Cfg["subnet"]["_"].Get("proxy")
+	callhomeCfg := result.Cfg[config.CallhomeSubSys]["_"]
+	if result.SubSys == config.CallhomeSubSys && callhomeCfg.Get(callhome.Enable) == "on" {
+		loggerWebhookSubnetProxy := result.Cfg[config.LoggerWebhookSubSys][config.SubnetSubSys].Get(logger.Proxy)
+		subnetProxy := result.Cfg[config.SubnetSubSys]["_"].Get(logger.Proxy)
 		if loggerWebhookSubnetProxy == "" && subnetProxy != "" {
-			obj := result.Cfg["logger_webhook"]["subnet"]
-			obj.Set("proxy", subnetProxy)
+			obj := result.Cfg[config.LoggerWebhookSubSys][config.SubnetSubSys]
+			obj.Set(logger.Proxy, subnetProxy)
 			result.LoggerWebhookCfgUpdated = true
 		}
 	}

--- a/internal/config/subnet/subnet.go
+++ b/internal/config/subnet/subnet.go
@@ -32,6 +32,8 @@ import (
 
 const (
 	respBodyLimit = 1 << 20 // 1 MiB
+
+	LoggerWebhookName = "subnet"
 )
 
 // Upload given file content (payload) to specified URL

